aleph/logic/notifications.py

```diff
@@ -66,7 +70,12 @@ def get_notifications(role, since=None, parser=None):
     filters = [{'terms': {'channels': channels}}]
     if since is not None:
         filters.append({'range': {'created_at': {'gt': since}}})
-    must_not = [{'term': {'actor_id': role.id}}]
+    ### TODO: this is a workaround for the notifictations http/500 on in single user mode.
+    ### Is this to ugly to exist?
```

Comment:
These work-arounds feel like they're addressing a secondary problem of some sort, rather than the root cause. What is the scenario where a user hits `/notifications` without the token being set?

Reply:
Agree that it feels wrong. Still unsure what would be correct.

>> What is the scenario where a user hits /notifications without the token being set?
Any request in SINGLE_USER mode ends in http/500

/ failed with status code 500 and:
|   File "/aleph/aleph/logic/notifications.py", line 78, in get_notifications
api_1               |     must_not = [{'term': {'actor_id': role.id}}]
api_1               | AttributeError: 'NoneType' object has no attribute 'id'

/search ends with:
DETAIL:  Key (role_id)=(26) is not present in table "role".
api_1               | [SQL: INSERT INTO query_log (query, session_id, role_id, created_at) VALUES (%(query)s, %(session_id)s, %(role_id)s, %(created_at)s) RETURNING query_lo

Reply:
Oooh I have a suspicion where this is going. Check what `Role.load_cli_user()` does, you may find that while it returns a valid user role it does not in fact commit the SQL transaction and thus the row is deleted after the `metadata` call returns. Maybe you just need to add `db.session.commit()` to the section where you add the token to the metadata.

Reply:
done